### PR TITLE
Test/waiver app

### PR DESCRIPTION
**Pull request by clmedders**
This PR is for the waiver page of the site. 
tests will cover

- sorting by alapa order 
- filter is working and the UI changes as expected 
- accordions are opening as expected 
- PSC code is present and just a number in the header and then in the body it matches what is in the header 
- the industry code is present in the description 
- OMB determination is correct based on the status of the waiver 

there are data attributes added to HTML files to help with testing files

### CODE REVIEW

**Comment by clmedders** [comment]
I didnt know for sure if I could delete files out of the pull request or if I need to create a new branch (which I can if needed) just didn't want to blow anything up but wanted to at least get this one test up.

Let me know if I need to re-do it!

File: cypress/integration/waiver-app_tests/filter-app.spec.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ /// <reference types="Cypress" /> 
+describe('filter by status', () => {
+it('UI changes when filter is selected', () => {
+  cy.visit('localhost:4000/waivers')
```
Comment (scottqueen-bixal) [inline cypress/integration/waiver-app_tests/filter-app.spec.js:4]: move to `before()` hook

File: cypress/integration/waiver-app_tests/pagination.spec.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ /// <reference types="Cypress" />
```
Comment (scottqueen-bixal) [inline cypress/integration/waiver-app_tests/pagination.spec.js:1]: remove this file from the PR

File: cypress/integration/waiver-app_tests/search_results.spec.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+ /// <reference types="Cypress" />
```
Comment (scottqueen-bixal) [inline cypress/integration/waiver-app_tests/search_results.spec.js:1]: remove this file from PR if not used

File: cypress/integration/waiver-app_tests/sort_app.spec.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+ /// <reference types="Cypress" />
```
Comment (scottqueen-bixal) [inline cypress/integration/waiver-app_tests/sort_app.spec.js:1]: remove this from the PR if not part of the work to be merged

File: search-app/src/app/app.component.html [hunk 5]
```diff
@@ -11,6 +11,7 @@ <h1 class="margin-0 font-family-sans">Nonavailability Waivers</h1>
       [label]="'Filter By Status'"
       [options]="filterOptions"
       (optionChange)="handleSelectedOptions($event)"
+      data-test="filter-select"
```
Comment (scottqueen-bixal) [inline search-app/src/app/app.component.html:14]: I like that we are starting to use these `data-test` attribs

**Comment by scottqueen-bixal** [comment]
try to avoid mixing separators in file names, ie `-` & `_`. You also may not need to include "test" since we know cypress is a testing framework.

consider  `component-or-app-being-tested/` +`compoent-or-functionality` so,  `integrations/search-app/filter-results.spec`

File: cypress/integration/waiver-app_tests/filter-app.spec.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  cy.get('[data-test="filter-select"] select').children().as('filter-options')
```
Comment (scottqueen-bixal) [inline cypress/integration/waiver-app_tests/filter-app.spec.js:5]: if you can move this alias to a `before()` hook, but after the `visit()` method, we should try to do that.

The idea of pattern being,

```
  before(() => {
    // include anything that might be part of the global test set up
  })
```

File: search-app/src/app/app.component.html (hunk 5 above)
Comment (clmedders) [reply search-app/src/app/app.component.html:14]: agreed... i think it will help make the tests better and more consistent!

**Comment by clmedders** [comment]
@scottqueen-bixal I have removed the sorting one till I can get that up to par.. the only ones that are in this PR are what was reviewed and given the thumbs-ups at code review!

**Comment by scottqueen-bixal** [comment]
when we blob the call for the npm `test` script we need to include the directory for the `waiver-app` test there so that all the tests under that directory get called.  One of the best ways to do that is to syntax the blob-like, `<directory>/**/*`.  I've gone ahead and updated to ensure when the pipeline is run it is running _all_ of our expected tests.

**Review by scottqueen-bixal — APPROVED** [review]
nice work @clmedders

**Pull request merged**